src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -533,26 +533,50 @@ private void RenderEndTagHelpersScope()
                                                   _tagHelperContext.ScopeManagerEndMethodName);
         }
 
-        private void RenderWriteTagHelperMethodCall(TagHelperChunk chunk)
+        private void RenderTagHelperOutput(TagHelperChunk chunk)
         {
+            var tagHelperOutputAccessor =
+                $"{ExecutionContextVariableName}.{_tagHelperContext.ExecutionContextOutputPropertyName}";
+
+            if (ContainsChildContent(chunk.Children))
```

Comment:
> I was saying "Why bother with this added complexity at all?" Still need to write output even if it's only whitespace.

Whitespace is still content, this if statement will be generated if there's whitespace. This statement is to prevent us creating a brand new `DefaultTagHelperContent` (which wont have anything in it) via `GetChildContentAsync` for self-closing tags that do nothing with their `output.Content`.

> I said self-closing was covered. It's Razor comments, complex statement blocks, templates, ... inside tag helpers that have poor coverage. No way to know if we're missing a case where Children contains something obscure that will no longer render. Again, why bother with these special cases?

We have suuuuuuper broad `TagHelper` tests across Razor and MVC repos to cover pretty much every scenario possible; I don't see the value in duplicating that here when this code generation addition affects every `TagHelper` scenario. 

Overall this `ContainsChildContent` check is very broad and returns `true` when in doubt, the only specific `Chunk` we check is `TagHelperChunk`. All other `ParentChunk`s are treated as pure containers (which they are in Razor) and we validate their children.